doc/netplan-tutorial.md

```diff
@@ -0,0 +1,560 @@
+
+# Getting started with Netplan
+
+In this tutorial we'll drive you through a number of practical exercises to get you started with Netplan. By the end of the tutorial you will be familiarized with some of the Netplan commands and configuration.
+
+## Pre-requisites
+
+In order to do the exercises yourself you will need a virtual machine, preferably running Ubuntu. In this tutorial, we will use LXD to create virtual networks and launch virtual machines. Feel free to use a cloud instance or a different hypervisor. As long as you can achieve the same results, you should be fine. If you're going to use your own desktop/laptop system, some of the exercises might interrupt your network connectivity.
+
+If you already have a setup where you can do the exercises you can just skip this section.
+
+### Setting up the environment
+
+You can follow the steps below to install and create a basic LXD configuration you can use to launch virtual machines. For more information about LXD, please visit: https://linuxcontainers.org/lxd/introduction/
+
+If you don't have LXD installed in your system, follow the steps below to install it:
+
+```bash
+$ snap install lxd
+$ lxd init --minimal
+```
+
+For some of the exercises you will need a second network interface in your virtual machine. Run the command below to create a new network in LXD:
+
+```bash
+lxc network create netplanbr0 --type=bridge
+```
+
+You should see the output below:
+
+```
+Network netplanbr0 created
+```
+
+After executing the commands above you will have a usable LXD installation with a working network bridge.
+
+Now create a virtual machine called `netplan-lab0`:
+
+```bash
+lxc init --vm ubuntu:22.04 netplan-lab0
+```
+
+You should see the out below:
+
+```
+Creating netplan-lab0
+```
+
+The new VM will have one network interface attached to the default LXD bridge.
+
+Now attach a second network interface to the VM and associate it to the LXD network you created earlier:
+
+```bash
+lxc network attach netplanbr0 netplan-lab0 eth1
+```
+
+And start your new VM:
+
+```bash
+lxc start netplan-lab0
+```
+
+Access your new VM using `lxc exec`:
+
+```bash
+lxc exec netplan-lab0 bash
```

Comment:
nitpick: `lxc shell netplan-lab0` should do the trick, too.

Maybe we should also mention the possibilty of `lxc console netplan-lab0` should somebody break their access to the VM.

Reply:
hmm interesting, lxc shell doesn't show up in the list of available commands for me... but it works